http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -95,53 +109,71 @@ void close() {
-   * failed.
+   * Initiates a non-blocking websocket connection.
    *
-   * @throws CouldNotConnect If the connection fails
+   * @param errorHandler Invoked if there is a failure to connect.
+   * @throws IllegalStateException Thrown if connection is already open (eg: connect called twice).
+   * @throws IllegalStateException Thrown if connection has been closed (ie: 'close()' was called)
    */
-  void connect() {
+  // Suppression: Ignoring future return values ignores any exceptions thrown by the code
+  // that completes the future. In this case we ignore the returned future from 'whenComplete'
+  // to avoid blocking. We would not expect the ping sender start or error handler to throw
+  // exceptions.
+  @SuppressWarnings("FutureReturnValueIgnored")
+  void connect(final Consumer<String> errorHandler) {
     Preconditions.checkState(!client.isOpen());
     Preconditions.checkState(!closed);
-    try {
-      client.connect();
-      pingSender.start();
-    } catch (final Exception e) {
-      pingSender.cancel();
-      throw new CouldNotConnect(serverUri);
-    }
+
+    connectAsync()
+        .whenComplete(
+            (connected, throwable) -> {
+              if (connected && throwable == null) {
+                pingSender.start();
+              } else {
+                errorHandler.accept("Failed to connect to: " + serverUri);
+              }
+            });
+  }
+
+  // Suppression: Ignoring future return values ignores any exceptions thrown by the code
+  // that completes the future. Connect blocking does not throw any exceptions, there are
+  // no exceptions to ignore, we can ignore the future returned by 'submit'.
+  @SuppressWarnings("FutureReturnValueIgnored")
+  private CompletableFuture<Boolean> connectAsync() {
+    final CompletableFuture<Boolean> completableFuture = new CompletableFuture<>();
+    // execute the connection attempt
+    Executors.newSingleThreadExecutor()
+        .submit(
+            () -> {
+              boolean connected;
+              try {
+                connected =
+                    client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
+              } catch (final InterruptedException ignored) {
+                connected = false;
+              }
+              return completableFuture.complete(connected);
+            });
+    return completableFuture;
   }
 
   /**
-   * Sends a message asynchronously.
+   * Sends a message asynchronously. Messages are queued until a connection has been established.
    *
-   * @throws IllegalStateException If the connection hasn't been opened yet. {@link #connect()}
-   *     needs to be called first.
+   * @throws IllegalStateException If the connection has been closed.
    */
   void sendMessage(final String message) {
     Preconditions.checkState(!closed);
-    if (!client.isOpen()) {
-      try {
-        Awaitility.await()
```

Comment:
Is the dependency now unused?

Reply:
Indeed, good catch, removed in: 18b0bacfd